src/python/WMComponent/WorkflowUpdater/WorkflowUpdaterPoller.py

```diff
@@ -0,0 +1,170 @@
+        self.rucioUrl = getattr(config.WorkflowUpdater, "rucioUrl",
+                                "http://cms-rucio-int.cern.ch")
+        self.rucioAuthUrl = getattr(config.WorkflowUpdater, "rucioAuthUrl",
+                                    "https://cms-rucio-auth-int.cern.ch")
+        self.rucioCustomScope = getattr(config.WorkflowUpdater, "rucioCustomScope", "group.wmcore")
+        self.msPileupUrl = getattr(config.WorkflowUpdater, "msPileupUrl",
+                                   "https://cmsweb-testbed.cern.ch/ms-pileup/data/pileup")
+
+        self.userCert = cert()
+        self.userKey = ckey()
+        self.rucio = Rucio(acct=self.rucioAcct,
+                           hostUrl=self.rucioUrl,
+                           authUrl=self.rucioAuthUrl)
+                           # configDict={'logger': self.logger})
+
+    @timeFunction
+    def algorithm(self, parameters=None):
+        """
+        Executed in every polling cycle. The actual logic of the component.
+        :param parameters: not really sure what those can be(!)
+        :return: only what is returned by the decorator
+        """
+        logging.info("Running Workflow updater injector poller algorithm...")
+        try:
+            # retrieve list of workflows with unfinished Production
+            # or Processing subscriptions
+            wflowSpecs = self.listActiveWflows.execute()
+            if not wflowSpecs:
+                logging.info("Agent has no active workflows at the moment")
+                return
+
+            # figure out workflows that have pileup
+            puWflows = self.findWflowsWithPileup(wflowSpecs)
+
+            # otherwise, move on retrieving pileups
+            pileupMap = self.getPileupDocs()
+        except Exception as ex:
+            msg = "Caught unexpected exception in WorkflowUpdater. Details:\n%s" % str(ex)
+            logging.exception(msg)
+            raise WorkflowUpdaterException(msg) from None
+
+    def getPileupDocs(self):
+        """
+        Fetch all pileup documents from MSPileup and preprocess the data.
+        :return: a list of dictionaries in the following format:
+          {"pileupName": string with pileup name,
+           "customName": string with custom pileup name - if any,
+           "rses": list of RSE names,
+           "blocks": list of block names}
+        """
+        mgr = RequestHandler()
+        headers = {'Content-Type': 'application/json'}
+        data = mgr.getdata(self.msPileupUrl, params={}, headers=headers, verb='POST',
+                           ckey=self.userKey, cert=self.userCert, encode=True, decode=True)
+        if data and data.get("result", []):
+            if "error" in data["result"][0]:
+                msg = f"Failed to retrieve MSPileup documents. Error: {data}"
+                raise WorkflowUpdaterException(msg)
+
+        logging.info("A total of %d pileup documents have been retrieved.", len(data["result"]))
+        pileupMap = []
```

Comment:
I suggest to change name of variable to `pileupMapList` as it more precisely represent its data-type.